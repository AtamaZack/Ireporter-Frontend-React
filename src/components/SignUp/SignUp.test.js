import React from 'react';
import { shallow } from 'enzyme';

import { SignUp, mapStateToProps } from './index';


describe('<SignUp />', () => {
  let wrapper;
  const props = {
    signUp: jest.fn()
  };
  beforeEach(()=>{
    wrapper = shallow(<SignUp {...props}/>);
  })

  it('should render the signup component', () => {  
    expect(wrapper).toMatchSnapshot();
  }); 

  it('handle change', ()=>{
    const e = {
      target:{
        name:"firstname",
        value:"zack"
      }
    }
    wrapper.instance().handleChange(e);
    expect(wrapper.instance().state.firstname).toBe('zack');
  });

  it('handle submit', ()=>{
    wrapper.instance().handleSubmit({ preventDefault:jest.fn() });
    expect(props.signUp).toHaveBeenCalled();
  });

  it('mapStateToProps', ()=>{
    const initialState = { signupReducer:{user: {},}
    };
    expect(mapStateToProps(initialState)).toEqual({user: {}})

  });

  it('mapDispatchToProps', ()=>{

  });

});
