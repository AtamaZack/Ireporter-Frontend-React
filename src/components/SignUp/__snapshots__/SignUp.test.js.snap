// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> should render the signup component 1`] = `ShallowWrapper {}`;
